### DISCUSSION TITLE
No more Bundle Product feature ?

### DISCUSSION THREAD

**Discussion by tlosbar** [saleor/saleor#9715] [General]
There was a backlog for implementing bundle products in the previous roadmap but it seems it disappears now. Do you abandon implementing this feature?

Because I started a workaround with the version of Saleor 3.0, it is working but it needs some tests, and I think my workaround is already deprecated with 3.1, 3.2, and of course 3.3.

Let me know if you plan to work on this or even if I can help with some PRs

**Comment by karolkielecki** [comment]
Hi @tlosbar, thanks for asking. We won't work on implementing product bundles anytime soon. We decided to update our roadmap and not put that feature on the list because we have other priorities for now. I think workarounds are possible. Could you please share more info on your workaround?

**Comment by rrrnld** [comment]
I would also be interested in a workaround! I was thinking of implementing something like this as a plugin, where you can name groups N products to be sold as bundles and the plugin would take care of keeping the stock of products that the bundles are made up off in sync. This is only a rough concept for now, I did not write any code yet.

**Reply by karolkielecki** [reply]
Hi @heyarne. Generally speaking, we do not recommend using plugins anymore for extending Saleor functionalities. So I'm not sure if that's a good approach. We recommend [apps](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts) instead of plugins.

**Reply by rrrnld** [reply]
Oh, that's news for me! Is there a rationale for this documented somewhere, a blog post or anything that goes into detail on this decision?

**Reply by karolkielecki** [reply]
There is no blog post about it, but we mentioned it a few times in similar discussions about extending Saleor. We believe that apps are best to extend Saleor capabilities. All new extensions should be apps.

**Comment by timuric** [comment]
@tlosbar you could create a product that would act as a bundle and link it to actual products with reference attributes. In order to calculate stock, you could use async webhooks that would do the math for bundles. All of this can be achieved without modifying the core or writing plugins.

Bundles might come back on the roadmap eventually, but so far most of our users are ok with custom implementations, due to fact that bundles can work very differently from business to business.
